cupy/cuda/cub.pyx

```diff
@@ -124,12 +124,20 @@ def device_reduce(ndarray x, op, out=None, bint keepdims=False):
     y_ptr = <void *>y.data.ptr
     dtype_id = _get_dtype_id(x.dtype)
     s = <Stream_t>stream.get_current_stream_ptr()
-
+    x_size = <int>x.size
```

Comment:
As a good Cython coding practice, could you declare `cdef int x_size` on the top?

Reply:
Or, if `with nogil:` below can be removed, you can simply revert to inlined `x.size` without casting.
How about removing `with nogil:`?

Reply:
I want to keep `with nogil` so I added `cdef int x_size`